honeybee_radiance/modifierset.py

```diff
@@ -0,0 +1,702 @@
+                for attr in attributes
+                }
+        else:
+            base = {
+                attr[1:]:getattr(self, attr[1:]).name
+                for attr in attributes
+            }
+        
+        base['type'] = self.__class__.__name__ + 'Abridged'
+        return base
+
+    def duplicate(self):
+        """Get a copy of this set."""
+        return self.__copy__()
+
+    def _validate_modifier(self, value):
+        """Check an modifier before assigning it."""
+        if value is not None:
+            assert isinstance(value, Primitive) and value.can_be_modifier , \
+                'Expected modifier. Got {}'.format(type(value))
+            # TODO: This should be uncommented once this is addressed
+            # 
+            # value.lock()   # lock editing in case modifier has multiple references
+        return value
+
+    @property
+    def _slots(self):
+        """Return slots including the ones from the baseclass if any."""
+        slots = set(self.__slots__)
+        for cls in self.__class__.__mro__[:-1]:
+            for s in getattr(cls, '__slots__', tuple()):
+                if s in slots:
+                    continue
+                slots.add(s)
+        slots.remove('_locked')
+        return slots
+
+    def ToString(self):
+        """Overwrite .NET ToString."""
+        return self.__repr__()
+
+    def __len__(self):
+        return 2
+
+    def __iter__(self):
+        return iter(self.modifiers)
+
+    def __copy__(self):
+        return self.__class__(self._exterior_modifier, self._interior_modifier)
+
+    def __repr__(self):
+        name = self.__class__.__name__.split('Set')[0]
+        return '{} Modifier Set:\n Exterior: {}\n Interior: {}'.format(
+            name, self.exterior_modifier.name, self.interior_modifier.name
+        )
+
+
+@lockable
+class WallSet(_BaseSet):
+    """Set containing all radiance modifiers needed to for an radiance model's Walls.
+
+    Properties:
+        exterior_modifier
+        interior_modifier
+        modifiers
+        modified_modifiers
+        is_modified
+    """
+    __slots__ = ()
+
+
+@lockable
+class FloorSet(_BaseSet):
+    """Set containing all radiance modifiers needed to for an radiance model's floors.
+
+    Properties:
+        exterior_modifier
+        interior_modifier
+        modifiers
+        modified_modifiers
+        is_modified
+    """
+    __slots__ = ()
+
+
+@lockable
+class RoofCeilingSet(_BaseSet):
+    """Set containing all radiance modifiers needed to for an radiance model's roofs.
+
+    Properties:
+        exterior_modifier
+        interior_modifier
+        modifiers
+        modified_modifiers
+        is_modified
+    """
+    __slots__ = ()
+
+
+@lockable
+class ApertureSet(_BaseSet):
+    """Set containing all radiance modifiers needed to for a radiance model's Apertures.
+
+    Properties:
+        exterior_modifier
+        interior_modifier
+        skylight_modifier
+        glass_door_modifier
```

Comment:
The more that I think about it what I would be doing if I were in practice, the more that I realize that I would find myself writing scripts to account for insect screens in operable windows such that they could be differentiated from fixed windows, which could be completely clear of the insect screens.  Just from looking at insect screens, it is clear that they block out at least as much light as another pane of glass would and, if we are already going through the trouble to make a distinction between an exterior double pane assembly and an interior single pane one, the insect screens seem to be just as relevant of an issue.

I also realize that this is a concern that is not relevant everywhere in the world since insect screens are more common in places with lots of flying insects, like the eastern US or China (I know that people mostly do not need them in Europe).

But I just want to go on the record saying that, if `operable_window_modifier` were included here in the `ModifierSet`, I would be one of the people using it. I don't know if I can really speak for others here so, I understand if you want to ask more people in the daylighting community before making a decision @mostaphaRoudsari .

Reply:
Let me re-think the categories for aperture / windows.

To me, the current logic that you are trying to come up with is backwards! If you are concerned about an extra screen just make the glass darker for those windows. I don't think you need an operable window category for that but I see your point.

Operable window can be interesting as the light transmittance will changed based on window being open or closed (or have insect screens or not). The question is if this should be addressed here or should be modeled as a Dynamic Window? I think the second one gives full control on how one wants to model the glass.

Also this is a default library for getting a minimum ready-to-go model. I think some of the complexity that you are talking about is out of the scope for default material set.